employees/forms.py

```diff
@@ -89,41 +89,35 @@ def __init__(self, *args: Any, **kwargs: Any):
         )
-        self._filter_task_activities_per_project()
-        if "data" not in kwargs:
-            self._set_last_choices_in_report_form(author)
-
-    def _filter_task_activities_per_project(self) -> None:
-        if "project" in self.data:
-            with suppress(ValueError, TypeError):
-                project_id = int(self.data.get("project"))
-                self.fields["task_activities"].queryset = TaskActivityType.objects.filter(projects=project_id).order_by(
-                    "name"
-                )
-        elif self.instance.pk:
-            self.fields["task_activities"].queryset = (
-                self.instance.project.project_activities.order_by("name")
-                | TaskActivityType.objects.filter(pk=self.instance.task_activities.pk)
-            ).distinct()
+        self._filter_task_activities(author)
 
-    def _set_last_choices_in_report_form(self, author: CustomUser) -> None:
+    def _filter_task_activities(self, author: CustomUser) -> None:
         if self.instance.pk is None:
             last_report = author.report_set.order_by("-creation_date").first()
             if last_report and (
                 author in last_report.project.members.all() or author in last_report.project.managers.all()
             ):
                 self.initial["project"] = last_report.project
-                self.fields["task_activities"].queryset = last_report.project.project_activities.all()
                 self.initial["task_activities"] = last_report.task_activities
+                self.fields["task_activities"].queryset = TaskActivityType.objects.filter(is_active=True)
             elif self.fields["project"].queryset.exists():
                 self.initial["project"] = self.fields["project"].queryset.first()
-                self.fields["task_activities"].queryset = self.initial["project"].project_activities.all()
+                self.fields["task_activities"].queryset = TaskActivityType.objects.filter(is_active=True)
             else:
                 self.fields["task_activities"].queryset = TaskActivityType.objects.none()
         else:
+            current_report = get_object_or_404(Report, pk=self.instance.pk)
+            self.initial["project"] = current_report.project
             self.fields["project"].queryset = (
                 self.fields["project"].queryset | Project.objects.filter(pk=self.instance.project.pk).distinct()
             )
+            self.initial["task_activities"] = current_report.task_activities
+            if current_report.task_activities in TaskActivityType.objects.filter(is_active=True):
+                self.fields["task_activities"].queryset = TaskActivityType.objects.filter(is_active=True)
+            else:
+                self.fields["task_activities"].queryset = TaskActivityType.objects.filter(
+                    is_active=True
+                ) | TaskActivityType.objects.filter(id=current_report.task_activities.pk)
```

Comment:
This method is so big that is shouting to be split :) I see that there are lot of duplicated code here so probably it will be simplified once you split it for separate well named methods.

Reply:
Changed function name to  _set_choices_for_fields_in_report_from as it filters not only task activities but also projects.  Two new functions added: _set_last_choices_while_creating_new_report_form and _set_choices_from_edited_report. Changed fields["task_activities"] in _set_last_choices_while_creating_new_report_form, i.e. always all active task activities will be given when creating new report (this imposed change in test_create_report_form_should_have_empty_project_and_activities_fields_if_user_is_not_in_any_active_project to test_create_report_form_should_have_empty_project_if_user_is_not_in_any_active_project)